Java/core/src/main/java/com/amazon/randomcutforest/state/sampler/CompactSamplerState.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.state.sampler;
+
+import java.util.Arrays;
+
+import lombok.Data;
+
+import com.amazon.randomcutforest.sampler.CompactSampler;
+
+@Data
+public class CompactSamplerState {
+
+    public float[] weightArray;
+
+    public int[] referenceArray;
+
+    public long[] sequenceArray;
+
+    public int size;
+
+    public int capacity;
+
+    public CompactSamplerState() {
+
+    }
+
+    public CompactSamplerState(CompactSampler sampler) {
```

Comment:
remark. it seems like storeSequence should be used here storing model into state.

Reply:
storeSequence really is a forest configuration option, because if it's enabled then sequences are propagated through a bunch of different places in the library. I think it's different than a saveTreeData flag as part of serialization, because in that case there's a tradeoff between between being able to restore the exact tree structure (at the cost of a larger serialized object), versus creating a smaller serialized object and then getting back a tree that's equal in distribution only. Throwing away sequence indexes during serialization would be a lossy operation.